nemo/collections/asr/models/classification_models.py

```diff
@@ -150,58 +158,102 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
         featurizer = WaveformFeaturizer(
             sample_rate=config['sample_rate'], int_values=config.get('int_values', False), augmentor=augmentor
         )
-
-        if 'vad_stream' in config and config['vad_stream']:
-            print("Perform streaming frame-level VAD")
-            dataset = AudioToSpeechLabelDataSet(
-                manifest_filepath=config['manifest_filepath'],
-                labels=config['labels'],
+        shuffle = config['shuffle']
+
+        # Instantiate tarred dataset loader or normal dataset loader
+        if config.get('is_tarred', False):
+            if ('tarred_audio_filepaths' in config and config['tarred_audio_filepaths'] is None) or (
+                'manifest_filepath' in config and config['manifest_filepath'] is None
+            ):
+                logging.warning(
+                    "Could not load dataset as `manifest_filepath` is None or "
+                    f"`tarred_audio_filepaths` is None. Provided config : {config}"
+                )
+                return None
+
+            if 'vad_stream' in config and config['vad_stream']:
+                logging.warning("VAD inference does not support tarred dataset now")
+                return None
+
+            shuffle_n = config.get('shuffle_n', 4 * config['batch_size']) if shuffle else 0
+            dataset = audio_to_label_dataset.get_tarred_classification_label_dataset(
                 featurizer=featurizer,
-                max_duration=config.get('max_duration', None),
-                min_duration=config.get('min_duration', None),
-                trim=config.get('trim_silence', True),
-                load_audio=config.get('load_audio', True),
-                time_length=config.get('time_length', 0.31),
-                shift_length=config.get('shift_length', 0.01),
-            )
-            batch_size = 1
-            collate_func = dataset.vad_frame_seq_collate_fn
-        else:
-            dataset = AudioLabelDataset(
-                manifest_filepath=config['manifest_filepath'],
-                labels=config['labels'],
-                featurizer=featurizer,
-                max_duration=config.get('max_duration', None),
-                min_duration=config.get('min_duration', None),
-                trim=config.get('trim_silence', True),
-                load_audio=config.get('load_audio', True),
+                config=config,
+                shuffle_n=shuffle_n,
+                global_rank=self.global_rank,
+                world_size=self.world_size,
+                augmentor=augmentor,
```

Comment:
augmentor is passed as an argument here but we do not do anything with it in get_tarred_classification_label_dataset()